internal/store/hpa.go

```diff
@@ -143,6 +144,57 @@ var (
 				}
 			}),
 		},
+		{
+			Name: "kube_hpa_status_currentmetrics_average_value",
+			Type: metric.Gauge,
+			Help: "Average metric value observed by the autoscaler.",
+			GenerateFunc: wrapHPAFunc(func(a *autoscaling.HorizontalPodAutoscaler) *metric.Family {
+				ms := make([]*metric.Metric, len(a.Status.CurrentMetrics))
+				for i, c := range a.Status.CurrentMetrics {
+					var value *resource.Quantity
+					switch c.Type {
+					case autoscaling.ResourceMetricSourceType:
+						value = &c.Resource.CurrentAverageValue
+					case autoscaling.PodsMetricSourceType:
+						value = &c.Pods.CurrentAverageValue
+					case autoscaling.ObjectMetricSourceType:
+						value = c.Object.AverageValue
+					case autoscaling.ExternalMetricSourceType:
+						value = c.External.CurrentAverageValue
+					default:
+						// Skip unsupported metric type
+						continue
+					}
+					intVal, canFastConvert := value.AsInt64()
+					if canFastConvert {
```

Comment:
It can be further condensed to this.
```suggestion
					if intVal, canFastConvert := value.AsInt64(); canFastConvert {
```